File: apps/api/app/schemas/dealership.py
import uuid
from collections import defaultdict
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DealershipResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    slug: str
    website_url: str | None
    status: str
    organization_id: uuid.UUID | None
    created_at: datetime


class DealershipUpdate(BaseModel):
    name: str | None = None
    website_url: str | None = None
    status: str | None = None


class RooftopGroupResponse(BaseModel):
    organization_id: uuid.UUID | None
    organization_name: str
    dealerships: list[DealershipResponse]


class AccessibleDealershipsResponse(BaseModel):
    groups: list[RooftopGroupResponse]
    active_organization_id: uuid.UUID | None
    active_dealership_id: uuid.UUID

File: apps/web/src/lib/types/dealership.ts
export interface Dealership {
  id: string;
  name: string;
  slug: string;
  website_url: string | null;
  status: string;
  organization_id: string | null;
  created_at: string;
}

export interface RooftopGroup {
  organization_id: string | null;
  organization_name: string;
  dealerships: Dealership[];
}

export interface AccessibleDealershipsResponse {
  groups: RooftopGroup[];
  active_organization_id: string | null;
  active_dealership_id: string;
}
